Lib/test/test_tarfile.py

```diff
@@ -3709,6 +3785,72 @@ def test_parent_symlink(self):
             with self.check_context(arc.open(), 'data'):
                 self.expect_file('parent/evil')
 
+    @symlink_test
+    def test_realpath_limit_attack(self):
+        # (CVE-2025-4517)
+        with ArchiveMaker() as arc:
+            # populate the symlinks and dirs that expand in os.path.realpath()
+            # The number is chosen so that in common cases, the unexpanded
+            # path fits in PATH_MAX, but it overflows when the final symlink
+            # is expanded
+            if sys.platform == 'darwin':
+                component = 'd' * 55
+            elif sys.platform == 'win32':
+                component = 'd' * 25
+            else:
+                component = 'd' * 247
+            steps = "abcdefghijklmnop"
+            path = ""
+            step_path = ""
+            for i in steps:
+                arc.add(os.path.join(path, component), type=tarfile.DIRTYPE,
+                        mode='drwxrwxrwx')
+                arc.add(os.path.join(path, i), symlink_to=component)
+                path = os.path.join(path, component)
+                step_path = os.path.join(step_path, i)
+            # create the final symlink that exceeds PATH_MAX and simply points
+            # to the top dir.
+            # this link will never be expanded by
+            # os.path.realpath(strict=False), nor anything after it.
+            linkpath = os.path.join(*steps, "l"*254)
+            arc.add(linkpath, symlink_to=os.path.join(*".." * len(steps)))
```

Comment:
currently it just is a chain of current-directory specifiers
```suggestion
            parent_segments = [".."] * len(steps)
            arc.add(linkpath, symlink_to=os.path.join(*parent_segments))
```